Graph/Basic Graph Traversal/133.py

```diff
@@ -0,0 +1,49 @@
+"""
+# Definition for a Node.
+class Node:
+    def __init__(self, val = 0, neighbors = None):
+        self.val = val
+        self.neighbors = neighbors if neighbors is not None else []
+"""
+
+# BFS Solution
+class Solution:
+    def cloneGraph(self, node: 'Node') -> 'Node':
+        if not node:
+            return node
+
+        graph = {}
+        queue = deque([node])
+        graph[node] = Node(node.val, [])
+
+        while queue:
+            n = queue.popleft()
+
+            for neighbor in n.neighbors:
+                if neighbor not in graph:
```

Comment:
Do you recognize the **_role_** this `graph` variable plays here (other than storing the cloned vertices)?

Reply:
It works as `seen`.

Reply:
Great! This is exactly what I wanted to hear.